main.tf

```
@@ -14,11 +14,151 @@
  * limitations under the License.
  */
 
-terraform {
-  required_version = "~> 0.12.0"
+locals {
+  seed_project_id             = format("%s-%s-%s", var.project_prefix, "seed", random_id.suffix.hex)
+  impersonation_apis          = distinct(concat(var.activate_apis, ["serviceusage.googleapis.com", "iamcredentials.googleapis.com"]))
+  impersonation_enabled_count = var.sa_enable_impersonation == true ? 1 : 0
+  activate_apis               = var.sa_enable_impersonation == true ? local.impersonation_apis : var.activate_apis
 }
 
-resource "google_storage_bucket" "main" {
-  project = var.project_id
-  name    = var.bucket_name
+resource "random_id" "suffix" {
+  byte_length = 2
 }
+
+data "google_organization" "org" {
+  organization = var.organization_id
+}
+
+/******************************************
+  Create IaC Project
+*******************************************/
+
+resource "google_project" "seed_project" {
+  name                = local.seed_project_id
+  project_id          = local.seed_project_id
+  org_id              = var.organization_id
+  billing_account     = var.billing_account
+  auto_create_network = false
+  depends_on = [
+    google_organization_iam_member.org_admins_group,
+  ]
+}
+
+resource "google_project_service" "seed_project_api" {
+  count                      = length(local.activate_apis)
```

This should use `for_each` to allow for nondestructive changes in the APIs list.

```suggestion
  for_each = toset(local.activate_apis)
  
```